core/alsp_src/tests/libtests/arithx1_tests.pro

```diff
@@ -0,0 +1,131 @@
+
+
+
+ta :- test_arithx1.
+
+test_arithx1 :-
+	test_max,
+	test_min,
+	test_maximum,
+	test_minimum,
+	test_sumlist,
+	test_prodlist,
+	test_sum_squares,
+	test_sum_square_diffs,
+	test_max_vector,
+	test_min_vector.
+
+
+test_max :-
+	max(3,92,M1),
+	M1 == 92,
+	max(3,-92,M2),
+	M2 == 3,
+	max(-6,-9,M3),
+	M3 == -6,
+	max(23.89, 23.87,M4),
+	M4 == 23.89,
+	!.
+	
+test_max :-
+	printf(user, 'test_max test failed\n', []).
+
+
+test_min :-
+	min(3,92,M1),
+	M1 == 3,
+	min(3,-92,M2),
+	M2 == -92,
+	min(-6,-9,M3),
+	M3 == -9,
+	min(23.89, 23.87,M4),
+	M4 == 23.87,
+	!.
+	
+test_min :-
+	printf(user, 'test_min test failed\n', []).
+
+test_maximum :-
```

Comment:
I assume you mean that the expression lists in the tests should contain compound (+, *, etc) expressions.

Reply:
Yea, they should contain arithmetic expressions, but more then that, they should explicitly test the bug you found. Maximum/minimum have always accepted arithmetic expressions, but the bug (I infer from the fix) is that the return value wasn't always a number. For example, the old maximum did this:

```
?- maximum([1, 1 + 1], X).
X = 1 + 1
```

While `maximum` has correctly identified the max item in the list, the result is unexpected (at least, compared to `max/3`). With the new `maximum`, the result is the more reasonable `X = 2`. An example test for this would be:

```
maximum([1 + 1], 2).
```

That fails with the old `maximum`, but succeeds with the new `maximum`, thus "proving" the bug fix worked (at least for that example).

The bug fix is tested in `test_minimum` now, so I'll mark resolved.